### PR TITLE
docs: add no-nesting examples

### PR DESCRIPTION
**Pull request by ChristianMurphy**
**What is the purpose of this pull request?**

* [x] Documentation update

**What changes did you make? (Give an overview)**

Added valid and invalid examples to `no-nesting` rule documentation

### CODE REVIEW

File: docs/rules/no-nesting.md [hunk 1]
```diff
@@ -1 +1,38 @@
...
+  .then(doSomethingElse)
+  .catch(errors)
+```
+
+#### Invalid
+
+```js
+myPromise
+  .then((val) => {
+    doSomething(val))
```
Comment (macklinu) [inline docs/rules/no-nesting.md]: Extra parenthesis at the end

File: docs/rules/no-nesting.md [hunk 2]
```diff
@@ -1 +1,38 @@
...
+      .then(doSomethingElse)
+  })
+
+myPromise
+  .then((val) => {
+    doSomething(val))
```
Comment (macklinu) [inline docs/rules/no-nesting.md]: Extra parenthesis at the end

File: docs/rules/no-nesting.md [hunk 3]
```diff
@@ -1 +1,38 @@
...
+      .catch(errors)
+  })
+
+myPromise
+  .catch((err) => {
+    doSomething(err))
```
Comment (macklinu) [inline docs/rules/no-nesting.md]: Extra parenthesis at the end

File: docs/rules/no-nesting.md [hunk 4]
```diff
@@ -1 +1,38 @@
...
+      .then(doSomethingElse)
+  })
+
+myPromise
+  .catch((err) => {
+    doSomething(err))
```
Comment (macklinu) [inline docs/rules/no-nesting.md]: Extra parenthesis at the end

File: docs/rules/no-nesting.md (hunk 2 above)
Comment (ChristianMurphy) [reply docs/rules/no-nesting.md]: removed and formatted with prettier

File: docs/rules/no-nesting.md (hunk 3 above)
Comment (ChristianMurphy) [reply docs/rules/no-nesting.md]: removed and formatted with prettier

File: docs/rules/no-nesting.md (hunk 4 above)
Comment (ChristianMurphy) [reply docs/rules/no-nesting.md]: removed and formatted with prettier

File: docs/rules/no-nesting.md (hunk 1 above)
Comment (ChristianMurphy) [reply docs/rules/no-nesting.md]: removed and formatted with prettier

**Pull request merged**